### PR TITLE
IaC for Dynatrace alerting config

### PR DESCRIPTION
**Pull request by chrisdodd93**
# Description:
IaC for alerting profiles and problem notifications

## Ticket number:
[OBS-199]

## Checklist:
- [ ] Is my change backwards compatible? Please include evidence
- [ ] I have tested this and added output to Jira Comment:
- [ ] Documentation added (link) Comment:

[OBS-199]: https://govukverify.atlassian.net/browse/OBS-199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: alerting_profiles.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            severity_level   = "ERRORS"
+        }
+        rule {
+            include_mode     = "NONE"
+            delay_in_minutes = 0
+            severity_level   = "MONITORING_UNAVAILABLE"
+        }
+        rule {
+            include_mode     = "NONE"
+            delay_in_minutes = 0
```
Comment (Pave1s) [inline alerting_profiles.tf:27]: current settings 30 mins.

File: alerting_profiles.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            severity_level   = "PERFORMANCE"
+        }
+        rule {
+            include_mode     = "NONE"
+            delay_in_minutes = 0
```
Comment (Pave1s) [inline alerting_profiles.tf:32]: same for this. 
current settings 30 mins.

**Review by Pave1s — CHANGES_REQUESTED** [review]

File: alerting_profiles.tf (hunk 1 above)
Comment (chrisdodd93) [reply alerting_profiles.tf:27]: Dynatrace default for resource contention and response time degradation is 30mins, although I'm not entirely sure why you'd want to be alerted 30mins after there has been a large spike in CPU or Memory or when your service has reduced performance and is operating slowly.
All this change means is that, when there is an issue, there won't be a delay in sending an alert into Slack. Alerts themselves are still controlled by the 3-in-5 rule, meaning three 1 minute periods in a 5 minute window will need to be above the threshold before firing an alert.
I'm not sure what the contractual agreement is with regards Live Service response times, but at the Home Office it was 30 minutes for a P1 incident; meaning in this circumstance you'd be responding outside of your contractual agreement to respond to an alert.

File: alerting_profiles.tf (hunk 2 above)
Comment (chrisdodd93) [reply alerting_profiles.tf:32]: Comment above applies here as well.

**Review by Pave1s — APPROVED** [review]

**Pull request merged**